src/components/ui/Modal/index.tsx

```diff
@@ -24,22 +24,25 @@ export interface ModalProps
   size?: ModalSize;
   /** The rootId of the modal. */
   rootId?: string;
+  /** Optional prop to prevent the modal from closing. */
+  dismissible?: boolean;
 }
 
 export const Modal: FC<ModalProps> = ({
   size = 'md',
   onClose,
   children,
   rootId,
+  dismissible = false,
   ...rest
 }) => {
   const labelID = useUniqueId();
   const contentEl = useRef<HTMLDivElement>(null);
   const mainEl = useRef<HTMLDivElement>(null);
 
-  const modalContext = { onClose, labelID };
+  const modalContext = { onClose, labelID, dismissible };
 
-  useClickOutside(mainEl, onClose);
+  !dismissible && useClickOutside(mainEl, onClose);
```

Comment:
Shouldn't `useClickOutside` work if `dismissable` is true?

Reply:
Updated